src/shared/logger/activation.ts

```diff
@@ -168,23 +168,36 @@ async function createLogWatcher(logFile: vscode.Uri): Promise<vscode.Disposable>
         return { dispose: () => {} }
     }
 
-    let checking = false
-    // TODO: fs.watch() has many problems, consider instead:
-    //   - https://github.com/paulmillr/chokidar
-    //   - https://www.npmjs.com/package/fb-watchman
-    const watcher = fs.watch(logFile.fsPath, async eventType => {
-        if (checking || eventType !== 'rename') {
-            return
-        }
-        checking = true
-        if (!(await SystemUtilities.fileExists(logFile))) {
-            vscode.window.showWarningMessage(
-                localize('AWS.log.logFileMove', 'The log file for this session has been moved or deleted.')
-            )
-            watcher.close()
-        }
-        checking = false
-    })
+    const disposables: vscode.Disposable[] = []
+
+    // Notify if log file is renamed
+    disposables.push(
+        vscode.workspace.onDidRenameFiles((e: vscode.FileRenameEvent) => {
```

Comment:
Nice change. Not a blocker for this PR, but we should consider dropping the log file watcher entirely. Instead just handle the failure if writing to the logfile fails. No need to create a watcher for this edge case...

Reply:
Sounds good, created IDE-11538